restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -589,6 +595,65 @@ private void invite(final Object message) throws IOException, NumberParseExcepti
 
     }
 
+    /**
+     * FIXME: duplicated code make into static function or something more optimized
+     * Replace headers
+     * @param SipServletRequest message
+     * @param Map<String, ArrayList<String> > headers
+     */
+    private void addHeadersToMessage(SipServletRequest message, Map<String, ArrayList<String> > headers) {
```

Comment:
@deruelle @scottbarstow should we do any further restrictions on modification of System headers, non system headers and custom headers?
We are only checking for two cases here, if a header is a Request-URI or not. 
System headers and non System headers will all equally be modified given any headers that we have defined in the db config.

An interesting scenario is if we have contending rcml headers and extension headers, have not tried this scenario. We can definitely do a string to check to completely exclude any custom 'X-' header modification.